include/RAJA/pattern/teams/teams_core.hpp

```diff
@@ -248,22 +269,135 @@ template <typename POLICY_LIST,
           typename CONTEXT,
           typename SEGMENT,
           typename BODY>
-RAJA_HOST_DEVICE RAJA_INLINE void loop(CONTEXT const &ctx,
+RAJA_HOST_DEVICE RAJA_INLINE void loop_idx(CONTEXT const &ctx,
                                        SEGMENT const &segment0,
                                        SEGMENT const &segment1,
                                        SEGMENT const &segment2,
                                        BODY const &body)
 {
 
 #if defined(RAJA_DEVICE_CODE)
-  LoopExecute<typename POLICY_LIST::device_policy_t, SEGMENT>::exec(
+  LoopIdxExecute<typename POLICY_LIST::device_policy_t, SEGMENT>::exec(
       ctx, segment0, segment1, segment2, body);
 #else
-  LoopExecute<typename POLICY_LIST::host_policy_t, SEGMENT>::exec(
+  LoopIdxExecute<typename POLICY_LIST::host_policy_t, SEGMENT>::exec(
       ctx, segment0, segment1, segment2, body);
 #endif
 }
```

Comment:
This is a style thing, but perhaps make this more like:

```
#If defined(RAJA_DEVICE_CODE)
using pol = typename POLICY_LIST::device_policy_t;
#else 
using pol = typename POLICY_LIST::host_policy_t;
#endif
```

This helps make it clear that they're doing the exact same thing, but with different policies, rather than replicating everything.

Reply:
Actually, looking through more, maybe even do this once higher up in the file and re-use it, or have a helper metafunction for it? There seem to be a lot of places with approximately this pattern, so it could cut down on the preprocessor checks by a good bit.

Reply:
I believe the last commit addresses this now. Thanks!

Reply:
Looks great!